src/launcher/java/org/truffleruby/launcher/options/CommandLineParser.java

```diff
@@ -50,7 +50,7 @@
 public class CommandLineParser {
 
     private final List<String> arguments;
-    private final List<String> jvmArguments;
+    private final List<String> argumentsToBeRemoved;
```

Comment:
How/when are they removed?
If this field still contains only jvm arguments or jvm+native arguments I think the name could be clearer.

Reply:
At the end of the argument parsing, options in this collection are removed from the original arguments. Currently it's only used for `-J*` options. (It can be found in the diff.)

I am not sure that it contains only `-J*` options matters much. The purpose is more interesting. If you don't find the name clear, suggest something. I've already thought about it and picked `argumentsToBeRemoved` to make it more clear.

Reply:
Maybe just `javaOptionsToTranslate` or `javaOptionsToBeRemoved` then.

Removing feels a bit strange for me, could we just accumulate them as we go and pass them in the same order (but translated to --jvm) to the AbstractLauncher?

Reply:
in-place replacement works nicely.